### ISSUE TITLE
Could not return discriminator key as Property in child class

### ISSUE THREAD

**Issue by duyhotan2000** [TypedProject/ts-express-decorators#1561]
Hi everybody,

I'm using Tsed with Mongoose and Model.
I defined a parent class:
```typescript
@Model()
export class BaseUser {
  @DiscriminatorKey()
  type: string;
}
```

And my child class:

```typescript
@Model({ discriminatorValue: UserType.ADMIN })
export class Admin extends BaseUser {
  @Property()
  specialProp: string;
}
```

I used this to filter the return fields on controller:

```typescript
@(Returns(200, Admin).Groups(UserType.ADMIN))
```
But it didn't return the `type` value because I could not define it as a Property, or I will get `Discriminator "admin" cannot have field with name "type"` error.

How do I get the `type` returned?

Thanks!

**Comment by Romakita** [comment]
Hello @duyhotan2000 
Ok produce a repository example. I don't use discriminator and I don't really understand the goals of this mongoose feature.

So repository example with the expected result will help me.

I won't investigate until this point is not ready ^^.


See you
Romain

**Comment by Romakita** [comment]
Hello @duyhotan2000 
I think it's fixed with this PR #1580 ^^

See you
Romain
